plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/BannedElementTest.java

```diff
@@ -29,6 +32,38 @@ public void testBannedElementConstructor_nullElementName() {
   public void testBannedElementConstructor_nullLocation() {
     new BannedElement("test", "org.eclipse.core.resources.problemmarker", 1, null, 0);
   }
+  
+  @Test
+  public void testEquals() {
+    BannedElement element1 = new BannedElement("message");
+    BannedElement element2 = new BannedElement("message");
+    assertTrue(element1.equals(element2) && element2.equals(element1));
+    
+    BannedElement element3 =
+        new BannedElement("message", "markerId_1", 1, new DocumentLocation(1, 1), 1);
```

Comment:
Pro tip: Use different constant values; e.g. 1, 2 rather than 1 and 1. That way if something fails, it's more obvious what went wrong. Also, if you use less obvious constants (28 and 73) it makes code search and debugging much easier when something goes wrong than if every piece of test data uses the same values.